src/main/java/bsh/classpath/ClassManagerImpl.java

```diff
@@ -121,8 +120,8 @@ This is lazily constructed and further (and more importantly) lazily
     private BshClassPath fullClassPath;
 
     // ClassPath Change listeners
-    private Vector listeners = new Vector();
-    private ReferenceQueue refQueue = new ReferenceQueue();
+    private Vector<WeakReference<Listener>> listeners = new Vector<>();
+    private ReferenceQueue<Listener> refQueue = new ReferenceQueue<>();
```

Comment:
I was wondering where the type parameters were. =)

The use of `Vector` vs `ArrayList` also needs to be considered here.